src/OrleansConsulUtils/SerializableMembershipTypes.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Consul;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Orleans.Runtime.Host
+{
+    /// <summary>
+    /// JSON Serializable Object that when serialized and Base64 encoded, forms the Value part of the Consul KVPair
+    /// </summary>
+    [JsonObject]
+    public class ConsulSiloRegistration
+    {
+        private static String DeploymentKVPrefix = "orleans";  //Ensures a root KV namespace in Consul
+        private static Char KeySeparator = '/';
+
+        //Internal properties are used to hold the KV.Key and CAS value
+        internal String DeploymentId { get; set; }  //Unique Key 
```

Comment:
I don't think it is necessary as that information is already held in the KV key.  If you would prefer it to be duplicated in the encoded Value I can easily add it.